python/hsfs/feature_view.py

```diff
@@ -122,46 +122,55 @@ def get_batch_query(
+                providing feature values which are not available in the feature store.
             external: boolean, optional. If set to True, the connection to the
                 online feature store is established using the same host as
                 for the `host` parameter in the [`hsfs.connection()`](project.md#connection) method.
                 If set to False, the online feature store storage connector is used
                 which relies on the private IP.
         # Returns
             `list` List of feature values related to provided primary keys, ordered according to positions of this
-            features in training dataset query.
+            features in the feature view query.
         """
         if self._vector_server is None:
             self.init_serving(external=external)
-        return self._vector_server.get_feature_vector(self, entry, external)
+        return self._vector_server.get_feature_vector(entry, passed_features)
 
     def get_feature_vectors(
-        self, entry: Dict[str, List[Any]], external: Optional[bool] = False
+        self,
+        entry: List[Dict[str, Any]],
+        passed_features: List[Dict[str, Any]],
+        external: Optional[bool] = False,
     ):
         """Returns assembled serving vectors in batches from online feature store.
 
         # Arguments
-            entry: dict of feature group primary key names as keys and value as list of primary keys provided by
-                serving application.
+            entry: a list of dictionary of feature group primary key and values provided by serving application.
+            passed_features: a list of dictionary of feature values provided by the application at runtime.
+                They can replace features values fetched from the feature store as well as
+                providing feature values which are not available in the feature store.
             external: boolean, optional. If set to True, the connection to the
                 online feature store is established using the same host as
                 for the `host` parameter in the [`hsfs.connection()`](project.md#connection) method.
                 If set to False, the online feature store storage connector is used
                 which relies on the private IP.
         # Returns
-            `List[list]` List of lists of feature values related to provided primary keys, ordered according to
-            positions of this features in training dataset query.
+            `List[list]` List of lists of feature values related to provided primary keys, ordered according to positions of this features in the feature view query.
         """
         if self._vector_server is None:
             self.init_serving(batch=True, external=external)
-        return self._vector_server.get_feature_vectors(self, entry, external)
+        return self._vector_server.get_feature_vectors(entry)
```

Comment:
`passed_features` argument is missing `self._vector_server.get_feature_vectors(entry, passed_features)`